### ISSUE TITLE
Passing `--noncritical` or `--skiponfailure` using `robot.run` API as a string is broken

### ISSUE THREAD

**Issue by thuibr** [robotframework/robotframework#3882]
#3682 breaks the `-noncritical` flag.

Seen in release 4.0, running Python 3.8:
```
[ ERROR ] Unexpected error: TypeError: can only concatenate list (not "tuple") to list
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.8/site-packages/robot/utils/application.py", line 83, in _execute
    rc = self.main(arguments, **options)
  File "/home/user/.local/lib/python3.8/site-packages/robot/run.py", line 447, in main
    result = suite.run(settings)
  File "/home/user/.local/lib/python3.8/site-packages/robot/running/model.py", line 292, in run
    self.visit(runner)
  File "/home/user/.local/lib/python3.8/site-packages/robot/model/testsuite.py", line 195, in visit
    visitor.visit_suite(self)
  File "/home/user/.local/lib/python3.8/site-packages/robot/model/visitor.py", line 88, in visit_suite
    suite.tests.visit(self)
  File "/home/user/.local/lib/python3.8/site-packages/robot/model/itemlist.py", line 77, in visit
    item.visit(visitor)
  File "/home/user/.local/lib/python3.8/site-packages/robot/model/testcase.py", line 103, in visit
    visitor.visit_test(self)
  File "/home/user/.local/lib/python3.8/site-packages/robot/running/suiterunner.py", line 123, in visit_test
    self._settings.skip_on_failure,
  File "/home/user/.local/lib/python3.8/site-packages/robot/conf/settings.py", line 490, in skip_on_failure
    return (self['SkipOnFailure'] or []) + (self['NonCritical'] or [])
```

Simplifying the problem:
```
>>> a = []
>>> b = ('noncritical', None)
>>> a + b
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate list (not "tuple") to list
```

**Comment by pekkaklarck** [comment]
I just tested that the basic usage works as expected:

**Comment by oboehmer** [comment]
Just ran into the same issue, this happens when you use the run API entrypoint, apparently the new skiponfailure option doesn't properly handles scalar values:

```
from robot import run
print('#### this one works:')
run('test.robot', skiponfailure=['non-critical'])
print('#### this one fails:')
run('test.robot', skiponfailure='non-critical')
```

running it triggers the issue:

(venv) OBOEHMER-M-C1EC:tmp oboehmer$ python run.py
#### this one works:
==============================================================================
Test
==============================================================================
test1                                                                 | PASS |
------------------------------------------------------------------------------
test2                                                                 | SKIP |
Test failed but its tags matched '--SkipOnFailure' and it was marked skipped.

Original failure:
this-is-a-test
------------------------------------------------------------------------------
test3                                                                 | PASS |
------------------------------------------------------------------------------
Test                                                                  | PASS |
3 tests, 2 passed, 0 failed, 1 skipped
==============================================================================
Output:  /private/tmp/output.xml
Log:     /private/tmp/log.html
Report:  /private/tmp/report.html
#### this one fails:
==============================================================================
Test
==============================================================================
[ ERROR ] Unexpected error: TypeError: can only concatenate str (not "list") to str
Traceback (most recent call last):
  File "/private/tmp/venv/lib/python3.8/site-packages/robot/utils/application.py", line 83, in _execute
    rc = self.main(arguments, **options)
  File "/private/tmp/venv/lib/python3.8/site-packages/robot/run.py", line 447, in main
    result = suite.run(settings)
  File "/private/tmp/venv/lib/python3.8/site-packages/robot/running/model.py", line 292, in run
    self.visit(runner)
  File "/private/tmp/venv/lib/python3.8/site-packages/robot/model/testsuite.py", line 195, in visit
    visitor.visit_suite(self)
  File "/private/tmp/venv/lib/python3.8/site-packages/robot/model/visitor.py", line 88, in visit_suite
    suite.tests.visit(self)
  File "/private/tmp/venv/lib/python3.8/site-packages/robot/model/itemlist.py", line 77, in visit
    item.visit(visitor)
  File "/private/tmp/venv/lib/python3.8/site-packages/robot/model/testcase.py", line 103, in visit
    visitor.visit_test(self)
  File "/private/tmp/venv/lib/python3.8/site-packages/robot/running/suiterunner.py", line 123, in visit_test
    self._settings.skip_on_failure,
  File "/private/tmp/venv/lib/python3.8/site-packages/robot/conf/settings.py", line 490, in skip_on_failure
    return (self['SkipOnFailure'] or []) + (self['NonCritical'] or [])
test1                                                                 (venv) OBOEHMER-M-C1EC:tmp oboehmer$
```

**Comment by pekkaklarck** [comment]
Thanks for the info @oboehmer. The problem you reported seems to be easy to fix and passing the value as a list is also an easy workaround The original descriptions uses `--noncritical` instead of `--skiponfailure`, but because the former is nowadays an alias for the latter I assume the root cause is the same. I need to test this myself but definitely sounds like something to be fixed in RF 4.0.1.

**Comment by oboehmer** [comment]
Thanks, Pekka! Just for the record, the same error is seen with `run('test.robot', noncritical='non-critical')`
